### PR TITLE
add APIs

### PR DESCRIPTION
**Pull request by ahmedali6**
This merge requests adds support to the following APIs:

### VirtualMachineService

- `importVm`
- `unmanageVirtualMachine`
- `listUnmanagedInstances`
- `importUnmanagedInstance`
- `createVMSchedule`
- `updateVMSchedule`
- `listVMSchedule`
- `deleteVMSchedule `

### ZoneService

- `listVmwareDcVms`
- `addVmwareDc`
- `listVmwareDcs`
- `updateVmwareDc`
- `removeVmwareDc`

### UserService

- `listUserData`
- `deleteUserData`
- `registerUserData`
- `moveUser`
- `setupUserTwoFactorAuthentication`

### ProjectService

- `createProjectRole`
- `updateProjectRole`
- `deleteProjectRole`

### RoleService

- `listProjectRoles`

### ObjectStoreService

- `createBucket`
- `deleteBucket`
- `updateBucket`
- `listBuckets`

### HostService

- `migrateSecondaryStorageData`
- `cancelHostAsDegraded`
- `listHostHAProviders`
- `listSecondaryStorageSelectors`
- `createSecondaryStorageSelector`
- `removeSecondaryStorageSelector`
- `listHostHAResources`
- `declareHostAsDegraded`
- `updateSecondaryStorageSelector`

### StoragePoolService

- `listObjectStoragePools`
- `listStoragePoolObjects`
- `updateObjectStoragePool`
- `addObjectStoragePool`
- `deleteObjectStoragePool`
- `listStoragePoolsMetrics`

### OauthService

- `listOauthProvider`
- `updateOauthProvider`
- `deleteOauthProvider`

### RegistrationService

- `registerOauthProvider`

### ShutdownService

- `readyForShutdown`

### VolumeService

- `listVolumesUsageHistory`
- `assignVolume`

### TemplateService

- `linkUserDataToTemplate`

### GuestOSService

- `getHypervisorGuestOsNames`

### ImageStoreService

- `listImageStoreObjects`
- `updateImageStore`
- `downloadImageStoreObject`

### PodService

- `updatePodManagementNetworkIpRange`

### ConfigurationService

- `updateStorageCapabilities`

### CertificateService

- `listCAProviders`
- `provisionCertificate`

### DiagnosticsService

- `getDiagnosticsData`

### MetricsService

- `listInfrastructure`

### ManagementService
 - `listManagementServers`

### CODE REVIEW

**Comment by ahmedali6** [comment]
@rohityadavcloud

File: generate/layout.go [hunk 1]
```diff
@@ -211,6 +211,7 @@ var layout = apiInfo{
 		"releaseHostReservation",
 		"updateHost",
 		"updateHostPassword",
+		"migrateSecondaryStorageData",
```
Comment (Pearl1594) [inline generate/layout.go:214]: Do you think this would be better to be placed in the ImageStoreService group?

Comment (ahmedali6) [reply generate/layout.go:214]: Hi, I'm just following the api refrence https://cloudstack.apache.org/api/apidocs-4.19/ where this is under Host, but yes I believe it's better placed in the ImageStoreService.

Comment (Pearl1594) [reply generate/layout.go:214]: Ah, got it. My bad. I guess it would make sense to correct it there as well (eventually).

**Comment by ahmedali6** [comment]
@weizhouapache Do you why the job failed? I only see `GitHub Actions has encountered an internal error when running your job.` and I can't see the logs.

File: cloudstack/HostService.go [hunk 2]
```diff
@@ -4753,3 +4772,1202 @@ func (r *UpdateHostPasswordResponse) UnmarshalJSON(b []byte) error {
...
+	return p
+}
+
+// Creates a secondary storage selector, described by the heuristic rule.
+func (s *HostService) CreateSecondaryStorageSelector(p *CreateSecondaryStorageSelectorParams) (*CreateSecondaryStorageSelectorResponse, error) {
+	resp, err := s.cs.newRequest("createSecondaryStorageSelector", p.toURLValues())
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println(string(resp))
```
Comment (vishesh92) [inline cloudstack/HostService.go:5540]: @ahmedali6 Can you remove this? Or run `make all` and push the synced changes.

**Comment by vishesh92** [comment]
@ahmedali6 Thanks for the PR. It looks good. Just one minor comment which needs to be resolved.

File: cloudstack/HostService.go (hunk 2 above)
Comment (vishesh92) [inline cloudstack/HostService.go:5540]:
```suggestion

```

Comment (ahmedali6) [reply cloudstack/HostService.go]: Fixed

Comment (ahmedali6) [reply cloudstack/HostService.go]: Fixed

**Comment by ahmedali6** [comment]
Hi. I will close this and make another one.

**Pull request closed without merging**